### PR TITLE
FastAPI integration

### PR DESCRIPTION
**Pull request by Skeen**
## Description

This PR introduces a shell integration around the ASGI integration.

It can be tested using `example.py`:
```Python
from typing import List

from uvicorn.main import run

import strawberry
from fastapi import FastAPI
from strawberry.fastapi import GraphQLRouter


@strawberry.type
class Book:
    title: str
    author: str


def get_books():
    return [
        Book(
            title="The Great Gatsby",
            author="F. Scott Fitzgerald",
        ),
    ]


@strawberry.type
class Query:
    books: List[Book] = strawberry.field(resolver=get_books)


schema = strawberry.Schema(Query)

app = FastAPI()
app.include_router(
    GraphQLRouter(schema, graphiql=True),
    prefix="/graphql",
    tags=["GraphQL"],
)
if __name__ == "__main__":
    run("example:app", port=5011, reload=True)
```

The main argument for existence of this shell integration is having GraphQL exposed in the `openapi.json` generated by FastAPI, and thus in the swagger / redoc documentation:

image

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Documentation

## Checklist

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Emil 'Skeen' Madsen for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: docs/integrations/fastapi.md [hunk 1]
```diff
@@ -4,21 +4,22 @@ title: FastAPI
 
 # FastAPI
 
-Strawberry provides support for [FastAPI](https://fastapi.tiangolo.com/) with the ASGI integration.
-
-See below example for integrating FastAPI with Strawberry:
+Strawberry comes with a [FastAPI](https://fastapi.tiangolo.com/) integration.
+It provides a router that you can use to serve your GraphQL schema:
```
Comment (patrick91) [inline docs/integrations/fastapi.md:8]: should we mention we need to run do `pip install strawberry-graphql[fastapi]`?

File: strawberry/fastapi/__init__.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+from fastapi.responses import HTMLResponse
+from fastapi.websockets import WebSocket
+from strawberry.asgi import GraphQL
+from strawberry.asgi.utils import get_graphiql_html
+from strawberry.http import GraphQLHTTPResponse, GraphQLRequestData, process_result
+from strawberry.utils.debug import pretty_print_graphql_operation
+
+
+NO_GRAPHIQL_MSG = "GraphiQL not enabled"
+GRAPHIQL_EXAMPLE = """
```
Comment (patrick91) [inline strawberry/fastapi/__init__.py:15]: can we move the these top level variables to different files?

File: docs/integrations/fastapi.md (hunk 1 above)
Comment (patrick91) [reply docs/integrations/fastapi.md:8]: thanks for updating the docs!

**Review by patrick91 — COMMENTED** [review]
This looks nice! Can we add some tests and restructure the code a bit? 

I think we should move away from having a lot of code inside `__init__.py` files. I've done that with ASGI but it's not great (I always find it hard to find what it)

File: strawberry/fastapi/__init__.py (hunk 2 above)
Comment (Skeen) [reply strawberry/fastapi/__init__.py:15]: Where would you like them? :)

Comment (patrick91) [reply strawberry/fastapi/__init__.py:15]: mmh

`GraphQLRouter` -> `fastapi/router.py`
`NO_GRAPHIQL_MSG` and `GRAPHIQL_EXAMPLE` -> `fastapi/constants.py`

Comment (Skeen) [reply strawberry/fastapi/__init__.py:15]: You bet!

**Comment by Skeen** [comment]
Closed in favor of #1291

**Pull request closed without merging**